pilot/pkg/xds/ads.go

```diff
@@ -321,6 +315,10 @@ func (s *DiscoveryServer) Stream(stream DiscoveryStream) error {
 	// reqChannel and the connection not being enqueued for pushes to pushChannel until the
 	// initialization is complete.
 	<-con.initialized
+	// authorize client
+	if err = s.authorize(con, ids); err != nil {
```

Comment:
This must happen before RegisterWorkload I feel like? Now its happening afterwards

Reply:
will check